plugins/out_s3/s3.c

```diff
@@ -1490,14 +1555,18 @@ static struct multipart_upload *get_upload(struct flb_s3 *ctx,
 
     mk_list_foreach_safe(head, tmp, &ctx->uploads) {
         tmp_upload = mk_list_entry(head, struct multipart_upload, _head);
-
+        if (tmp_upload->parts_uploaded == 0) {
+            mk_list_del(&m_upload->_head);
+            multipart_upload_destroy(m_upload);
+        }
```

Comment:
@Claych this will delete all upload which do not have parts uploaded yet... have you tested this? I think it will cause an issue. This bit of code should go down below in the if block where we check retry limit. The idea is to delete pending uploads which have repeatedly failed and for which no data (no parts) have been uplaoded.